scripts/encode_data.py

```diff
@@ -43,15 +48,46 @@ def embed_documents(self) -> None:
             if name == "data/links":
                 continue
 
-            thread = threading.Thread(target=self._process_document, args=(file,))
+            threading.Thread(target=self._process_document, args=(file,)).start()
```

Comment:
You need an else here, otherwise you'd spawn two separate threads here which I don't think youd want to do.